src/main/kotlin/org/datepollsystems/waiterrobot/mediator/printer/service/PrinterDiscoverService.kt

```diff
@@ -1,16 +1,36 @@
 package org.datepollsystems.waiterrobot.mediator.printer.service
 
+import org.datepollsystems.waiterrobot.mediator.App
+import org.datepollsystems.waiterrobot.mediator.app.Config
+import org.datepollsystems.waiterrobot.mediator.printer.AbstractLocalPrinter
 import org.datepollsystems.waiterrobot.mediator.printer.LocalPrinter
 import org.datepollsystems.waiterrobot.mediator.printer.LocalPrinterInfo
+import org.datepollsystems.waiterrobot.mediator.printer.VirtualLocalPrinter
 import javax.print.DocFlavor
 import javax.print.PrintServiceLookup
 
 object PrinterDiscoverService {
     // Get all printers that support printing PDF's in ByteArray format
-    val localPrinterMap: Map<String, LocalPrinter> = PrintServiceLookup
-        .lookupPrintServices(DocFlavor.SERVICE_FORMATTED.PAGEABLE, null)
-        .map { LocalPrinter(it) }
-        .associateBy { it.localId }
+    val localPrinterMap: Map<String, AbstractLocalPrinter> get() = _localPrinterMap
+    private var _localPrinterMap: Map<String, AbstractLocalPrinter> = emptyMap()
+
+    init {
+        refreshPrinters()
+    }
+
+    fun refreshPrinters() {
+        _localPrinterMap = PrintServiceLookup
+            .lookupPrintServices(DocFlavor.SERVICE_FORMATTED.PAGEABLE, null)
+            .map { LocalPrinter(it) }
+            .let {
+                if (App.config !is Config.Prod) {
+                    it.plus(VirtualLocalPrinter(1))
```

Comment:
Not really a fan of being only able to use the virtual printer on lava. But for the start it should be enough.

This also means that only one VirtualLocalPrinter is created and you are not able to add additional? Perhaps add a comment explaining this if it is the case.

Reply:
For now this will be a limitation. In the future (when there is time) we can add an UI for it. :)